File: src/front-end/fun-chess-app/src/core/chess/DetailedTeam.ts
import CastlingPlay from "./enums/CastlingPlay";
import Team from "./enums/Team";
import Position from "./structs/Position";

export default class DetailedTeam {
    public constructor(castlingPlays?: CastlingPlay, exposedEnPassant?: Position | null, team?: Team) {
        if (castlingPlays !== undefined && exposedEnPassant !== undefined) {
            this.castlingPlays = castlingPlays;
            this.exposedEnPassant = exposedEnPassant;

            return;
        }
        if (team !== undefined) {
            this.kingPosition = team === Team.White ? new Position(0, 4) : new Position(7, 4);
        }
    }

    public castlingPlays = CastlingPlay.All;

    public exposedEnPassant: Position | null = null;

    public kingPosition: Position | null = null;
}

File: src/back-end/FunChess/src/FunChess.Core.Chess/DetailedTeam.cs
using FunChess.Core.Chess.Enums;
using FunChess.Core.Chess.Structs;

namespace FunChess.Core.Chess;

public sealed class DetailedTeam
{
    public DetailedTeam(Team team)
    {
        KingPosition = team == Team.White ? new Position(0, 4) : new Position(7, 4);
    }
    
    internal DetailedTeam(CastlingPlay castlingPlays, Position? exposedEnPassant)
    {
        CastlingPlays = castlingPlays;
        ExposedEnPassant = exposedEnPassant;
    }
    
    public CastlingPlay CastlingPlays { get; internal set; } = CastlingPlay.All;

    public Position? ExposedEnPassant { get; internal set; }
    
    public Position? KingPosition { get; internal set; }
}